CMakePresets.json

```diff
@@ -0,0 +1,384 @@
+{
+    "version": 3,
```

Comment:
I am trying this right now on Linux with CLion, and so far I have observed that [CLion 2021.2.3 supports only version 2 Presets and only buildPresets](https://www.jetbrains.com/help/clion/cmake-presets.html).

When version is set to 3, as it is here, none of the presets are loaded. I am not fully convinced that this is a bad thing; CLion works well enough with this project without the presets. However, if we want this to work with CLion right now, this version number needs to come down to 2.

I would be surprised if CLion does not provide support for version 3 of CMakePresets.json at some point in the future, so there may be some value to leaving the version number as-is.

The CMake [docs](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html) describe the differences between versions 2 and 3; I did not see anything in the docs that made it look like we really need to use version 3.

Reply:
Lets create an issue to re-test this CMake presets implementation when CLion and QtCreator have a completed their support for this feature. Those should come very soon.

Reply:
CLion 2021.3.1 was released Dec. 1, and the docs say that it supports version 3. I haven't tested it yet, but at this point I don't think there's any reason to drop the version down to 2.

I'm not sure what's going on with QtCreator and CMake Presets. [The issue QTCREATORBUG-24555](https://bugreports.qt.io/browse/QTCREATORBUG-24555) was created in August 2020, and no progress has been noted in the bug tracker. 

I don't think we need to make sure that every IDE plays nice with this project. There are too many IDEs! I think VS Code is enough to worry about.

Reply:
Yes  I agree if we follow the de facto standard all the IDE will follow sooner or later!